src/Domain/Apprenticeship/EpisodeDomainModel.cs

```diff
@@ -0,0 +1,232 @@
+            {
+                shouldSupersedePreviousPrice = true;
+            }
+
+            var newEpisode = AddEpisodePrice(priceChangeRequest.EffectiveFromDate,
+                endDate,
+                priceChangeRequest.TotalPrice,
+                priceChangeRequest.TrainingPrice,
+                priceChangeRequest.AssessmentPrice,
+                fundingBandMaximum,
+                shouldSupersedePreviousPrice);
+            
+            return new AmendedPrices(newEpisode, _entity.Key, deletedPrices.ToList());
+        }
+
+        internal AmendedPrices UpdatePricesForApprovedStartDateChange(StartDateChangeDomainModel startDateChangeRequest, int fundingBandMaximum)
+        {
+            var latestPrice = LatestPrice;
+            var deletedPrices = DeletePricesEndingBeforeDate(startDateChangeRequest.ActualStartDate).ToList();
+            deletedPrices.AddRange(DeletePricesStartingAfterDate(startDateChangeRequest.PlannedEndDate));
+
+            if (ActiveEpisodePrices.Count == 0)
+            {
+                AddEpisodePrice(
+                    startDateChangeRequest.ActualStartDate,
+                    startDateChangeRequest.PlannedEndDate,
+                    latestPrice.TotalPrice,
+                    latestPrice.TrainingPrice,
+                    latestPrice.EndPointAssessmentPrice,
+                    fundingBandMaximum);
+            }
+            else
+            {
+                if (FirstPrice.StartDate != startDateChangeRequest.ActualStartDate)
+                {
+                    FirstPrice.UpdateStartDate(startDateChangeRequest.ActualStartDate);
+                }
+
+                if (LatestPrice.EndDate != startDateChangeRequest.PlannedEndDate)
+                {
+                    LatestPrice.UpdateEndDate(startDateChangeRequest.PlannedEndDate);
+                }
+            }
+
+            UpdateAllActivePricesWithNewFundingBandMaximum(fundingBandMaximum);
+
+            return new AmendedPrices(LatestPrice, _entity.Key, deletedPrices.ToList());
+        }
+
+        internal void UpdatePaymentStatus(bool isFrozen)
+        {
+            _entity.PaymentsFrozen = isFrozen;
+        }
+
+        public Episode GetEntity()
+        {
+            return _entity;
+        }
+
+        public static EpisodeDomainModel Get(Episode entity)
+        {
+            return new EpisodeDomainModel(entity);
+        }
+
+        public class AmendedPrices
+        {
+            public AmendedPrices(EpisodePriceDomainModel latestEpisodePrice, Guid apprenticeshipEpisodeKey, List<Guid> deletedPriceKeys)
+            {
+                DeletedPriceKeys = deletedPriceKeys;
+                LatestEpisodePrice = latestEpisodePrice;
+                ApprenticeshipEpisodeKey = apprenticeshipEpisodeKey;
+            }
+
+            public Guid ApprenticeshipEpisodeKey { get; set; }
+            public List<Guid> DeletedPriceKeys { get; set; }
```

Comment:
Do we use this deleted price keys variable anymore? Looks like it get's set but never read anywhere